tests/test_server_basic.py

```diff
@@ -184,6 +184,81 @@ def test_same_request_different_users(dispatcher_live_fixture):
     assert len(dir_list_1) == len(dir_list_2)
 
 
+@pytest.mark.not_safe_parallel
+@pytest.mark.parametrize("request_cred", ['public', 'private'])
+def test_consistency_parameters_json_dump_file(dispatcher_live_fixture, request_cred):
+    DispatcherJobState.remove_scratch_folders()
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+
+    if request_cred == 'public':
+        encoded_token = None
+    else:
+        token_payload = {
+            **default_token_payload,
+            "sub": "mtm@mtmco.net"
+        }
+
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    # issuing a request each, with the same set of parameters
+    params = {
+        'query_status': "new",
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token
+    }
+
+    jdata = ask(server,
+                  params,
+                  expected_query_status=["done"],
+                  max_time_s=50,
+                  )
+
+    assert jdata["exit_status"]["debug_message"] == ""
+    assert jdata["exit_status"]["error_message"] == ""
+    assert jdata["exit_status"]["message"] == ""
+
+    job_id = jdata['job_monitor']['job_id']
+    session_id = jdata['session_id']
+    # get the analysis_parameters json file
+    analysis_parameters_json_fn = f'scratch_sid_{session_id}_jid_{job_id}/analysis_parameters.json'
+    # the aliased version might have been created
+    analysis_parameters_json_fn_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased/analysis_parameters.json'
+    assert os.path.exists(analysis_parameters_json_fn) or os.path.exists(analysis_parameters_json_fn_aliased)
+    if os.path.exists(analysis_parameters_json_fn):
+        analysis_parameters_json_content_original = json.load(open(analysis_parameters_json_fn))
+    else:
+        analysis_parameters_json_content_original = json.load(open(analysis_parameters_json_fn_aliased))
+
+    logger.info("starting query with the same session_id and job_id")
+
+    # issue another call, different parameters but same job_id & session_id, to simulate the Fit button
+    params = {
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token,
+        'session_id': session_id,
+        'job_id': job_id,
+        'query_status': "ready"
+    }
+
+    jdata = ask(server,
+                params,
```

Comment:
but are you not also sending exactly the same parameters?

Reply:
no, I am just changing the 'query_status'

Reply:
so the situation as with fit is not happening, and even if the parameters were different it would not fail. And the special case of different parameters for the same job id, like with fit, is not tested.

Reply:
The `query_status` is used for the calculation of the job_id, so, from one test to the other, the parameters are different while having the same `job_id`. Unless I am missing a point

Reply:
`query_status` is **not** used in calculation of `job_id`:

https://github.com/oda-hub/dispatcher-app/blob/0657d9241c3980a888cf577a6943666b96f0d9b4/cdci_data_analysis/flask_app/dispatcher_query.py#L238-L245

And it really should not be. If it was, job_id would change when request progresses (e.g. to ready), and would be denied due to mismatching job_id

Reply:
damn now I see my mistake, I forgot to make a last pull, so my `restricted_par_dic` was not up to date, and `query_status` was not there.